src/lib.rs

```diff
@@ -45,15 +50,26 @@ pub async fn run(core_happ: &config::Happ, config: &config::Config) -> Result<()
     // suspend happs that have overdue payments
     let pending_transactions = get_pending_transactions(&mut core_app_client).await?;
     let suspended_happs = suspend_unpaid_happs(&mut core_app_client, pending_transactions).await?;
+    let hosting_preference = get_hosting_preferences(&mut core_app_client).await?;
 
     let list_of_happs = get_all_published_hosted_happs(&mut core_app_client).await?;
+    let mut publisher_jurisdictions: HashMap<String, Option<String>> = HashMap::new();
+    // get publisher jurisdiction for each happ
+    for happ in list_of_happs.iter() {
+        let happ_prefs = get_happ_preferences(&mut core_app_client, happ.happ_id.clone()).await?;
+        let publisher_jurisdiction =
+            get_publisher_jurisdiction(&mut core_app_client, happ_prefs.provider_pubkey).await?;
+        publisher_jurisdictions.insert(happ.happ_id.clone().to_string(), publisher_jurisdiction);
```

Comment:
You could maintain a list of publisher details that were already retrieved and then use it if their jurisdiction was already retrieved for a different happ